blake3-hasher/src/lib.rs

```diff
@@ -146,71 +65,19 @@ pub fn hash(val: &[u8]) -> Hash {
     hashv(&[val])
 }
 
-#[cfg(feature = "std")]
-/// Return the hash of the given hash extended with the given value.
-pub fn extend_and_hash(id: &Hash, val: &[u8]) -> Hash {
-    let mut hash_data = id.as_ref().to_vec();
-    hash_data.extend_from_slice(val);
-    hash(&hash_data)
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
 
     #[test]
-    fn test_new_unique() {
-        assert!(Hash::new_unique() != Hash::new_unique());
-    }
-
-    #[test]
-    fn test_hash_fromstr() {
-        let hash = hash(&[1u8]);
-
-        let mut hash_base58_str = bs58::encode(hash).into_string();
-
-        assert_eq!(hash_base58_str.parse::<Hash>(), Ok(hash));
-
-        hash_base58_str.push_str(&bs58::encode(hash.0).into_string());
-        assert_eq!(
-            hash_base58_str.parse::<Hash>(),
-            Err(ParseHashError::WrongSize)
-        );
-
-        hash_base58_str.truncate(hash_base58_str.len() / 2);
-        assert_eq!(hash_base58_str.parse::<Hash>(), Ok(hash));
-
-        hash_base58_str.truncate(hash_base58_str.len() / 2);
-        assert_eq!(
-            hash_base58_str.parse::<Hash>(),
-            Err(ParseHashError::WrongSize)
-        );
-
-        let input_too_big = bs58::encode(&[0xffu8; HASH_BYTES + 1]).into_string();
-        assert!(input_too_big.len() > MAX_BASE58_LEN);
-        assert_eq!(
-            input_too_big.parse::<Hash>(),
-            Err(ParseHashError::WrongSize)
-        );
-
-        let mut hash_base58_str = bs58::encode(hash.0).into_string();
-        assert_eq!(hash_base58_str.parse::<Hash>(), Ok(hash));
-
-        // throw some non-base58 stuff in there
-        hash_base58_str.replace_range(..1, "I");
-        assert_eq!(
-            hash_base58_str.parse::<Hash>(),
-            Err(ParseHashError::Invalid)
-        );
-    }
-
-    #[test]
-    fn test_extend_and_hash() {
+    fn test_hashv() {
         let val = "gHiljKpq";
         let val_hash = hash(val.as_bytes());
+
         let ext = "lM890t";
-        let hash_ext = [&val_hash.0, ext.as_bytes()].concat();
-        let ext_hash = extend_and_hash(&val_hash, ext.as_bytes());
+        let ext_hash = hashv(&[&val_hash.to_bytes(), ext.as_bytes()]);
```

Comment:
Will this work when the feature "offchain" is not enabled? Perhaps we need to put a feature cfg on the test?